File: hygiene_nerds_backend/src/main/java/com/universe/hygienenerds/hygiene_nerds_backend/dto/OrderConfirmationDto.java
package com.universe.hygienenerds.hygiene_nerds_backend.dto;

import lombok.Data;
import java.time.LocalDateTime;
import java.util.List;

@Data
public class OrderConfirmationDto {
    private String orderId;
    private String status;
    private String message;
    private LocalDateTime orderDate;
    private LocalDateTime estimatedDelivery;
    private List<OrderItemDto> items;
    private Double total;
    private ShippingAddressDto shippingAddress;
    private String paymentMethod;

    @Data
    public static class OrderItemDto {
        private String name;
        private Integer quantity;
        private Double price;
    }

    @Data
    public static class ShippingAddressDto {
        private String name;
        private String street;
        private String city;
        private String zipCode;
    }
}

File: hygiene_nerds_frontend/src/types/OrderConfirmation.dto.ts
export interface OrderConfirmationDto {
  orderId: string;
  status: "SUCCESS" | "PENDING" | "FAILED";
  message: string;
  orderDate: string;
  estimatedDelivery: string;
  items: {
    name: string;
    quantity: number;
    price: number;
  }[];
  total: number;
  shippingAddress: {
    name: string;
    street: string;
    city: string;
    zipCode: string;
  };
  paymentMethod: string;
}
